Fixed issues in ModelsTrainer and SpeakerIdentifier

manitbaser opened the pull request:
…se of SilenceEliminator; Achieved an accuracy of 98.132%

manitbaser committed:
Fixed issues in ModelsTrainer and SpeakerIdentifier; Eliminated the use of SilenceEliminator; Achieved an accuracy of 98.132%

SuperKogito commented on Code/ModelsTrainer.py:
```diff
@@ -34,12 +34,10 @@
-        silence_eliminator = SilenceEliminator()
 
         try   :
-            silence_eliminated_wave_file_path = "temp-" + os.path.basename(filepath).split('.')[0] + ".wav"
```
Silence eliminator is an important part, and should be kept. 
Maybe it is good idea here, to define an extra input variable specifying where to use silence eliminator or not.
Like for example a `bool` called `use_silence_eliminator`

SuperKogito reviewed (changes requested):
**Thank your for the PR. The changes made are valuable, especially the import edits.** 
However, **the silence eliminator should be kept**, as it is vital in such applications. The reason you use a VAD or here called a silence eliminator is because you want to base your model/ predictions on voice only data. Otherwise the model will take into account the "silent" noise of the speaker surroundings. Which could result in the following: If the recordings have similar "silent" noise then the prediction will be improved but in case the user speaks in a different setup resulting in a different surrounding noise/ silent noise then the prediction would not be accurate as the system will consider the silence instances as different to the ones the model is trained on. **So in conclusion, removing the silence eliminator might improve the accuracy in some cases, but overall it will degrade the robustness of the system.**

manitbaser committed:
Fixed errors in SilenceEliminator

manitbaser commented:
Hi, fixed the errors in SilenceEliminator and added it as discussed previously. SilenceEliminator wasn't running due to some bugs, so modified the file a bit.
